src/backends/perf_pt/collect.c

```diff
@@ -401,13 +415,14 @@ perf_pt_init_tracer(struct tracer_conf *tr_conf)
     tr_ctx->aux_buf = mmap(NULL, base_header->aux_size, PROT_READ | PROT_WRITE,
         MAP_SHARED, tr_ctx->perf_fd, base_header->aux_offset);
     if (tr_ctx->aux_buf == MAP_FAILED) {
+        perf_pt_set_err(err, perf_pt_cerror_errno, errno);
         failing = true;
         goto clean;
     }
 
 clean:
-    if (failing) {
-        perf_pt_free_tracer(tr_ctx);
+    if ((failing) && (tr_ctx != NULL)) {
+        perf_pt_free_tracer(tr_ctx, err);
```

Comment:
Here's the NULL pointer dereference fix.

Reply:
There's no need for the brackets around `(failing)` or `(tr_ctx != NULL)`.

Reply:
I agree with `failing` but the latter condition is easier to read with parens, isn't it? The whole code-base uses that style.

Reply:
It's a matter of taste, I admit, but since I'm not a Lisper, I rarely find that more brackets help. But if that's the style of this file, we might as well stick with it.